### DISCUSSION TITLE
including react?

### DISCUSSION THREAD

**Discussion by goern** [willfaught/paige#18] [General]
Heya, how would I go ahead, and include a react app under a certain path? I'd like to reuse the layout/design from the template...

**Comment by willfaught** [comment]
This might be more of a general Hugo question. I've never used React, so I'm afraid I can't be much help, and I'll be showing my ignorance.

In general, you can override the default layouts in layouts/_default (e.g. baseof.html) by making your own in layouts/mysection/mysubsection/, e.g. layouts/mysection/mysubsection/baseof.html. You could copy Paige's baseof.html and adapt it as needed.

If you want to run JavaScript in a particular page, you just need to &lt;script src=""&gt; include it (you'll need to enable unsafe markdown in config.yaml). You can put script tags into body-last.html (see the readme) if you want.

Try searching for "hugo site react". I found these, which might be helpful:

- https://stackoverflow.com/questions/53322953/hugo-with-react
- https://www.wictorwilen.se/blog/how-to-add-a-react-script-to-hugo/
- https://discourse.gohugo.io/t/integrating-react-in-hugo/35401
- https://discourse.gohugo.io/search?q=react

What concrete problem are you running into?

**Comment by goern** [comment]
Thanks for that. Next.JS seems to be the way, I'll explore...
